worker/codalabworker/worker.py

```diff
@@ -45,16 +45,21 @@ class Worker(object):
         4) Upgrading the worker.
     """
 
-    def __init__(self, id, tag, work_dir, max_work_dir_size_bytes, max_dependencies_serialized_length,
-                 max_images_bytes, shared_file_system,
-                 slots, bundle_service, docker, docker_network_prefix='codalab_worker_network'):
+    def __init__(self, id, tag, work_dir, cpuset, gpuset,
+                 max_work_dir_size_bytes, max_dependencies_serialized_length, max_images_bytes,
+                 shared_file_system, bundle_service, docker, docker_network_prefix='codalab_worker_network'):
         self.id = id
         self._tag = tag
         self.shared_file_system = shared_file_system
         self._bundle_service = bundle_service
         self._docker = docker
         self._docker_network_prefix = docker_network_prefix
-        self._slots = slots
+
+        self._resource_lock = threading.Lock() # lock for cpuset and gpuset
+        self._cpuset = cpuset
+        self._gpuset = gpuset
+        self._cpuset_free = set(cpuset)
```

Comment:
It's weird if `cpuset` is not a set...make it a set or call it `cpus`?

Reply:
_cpuset is in fact a set, and _cpuset_free is initialized as a copy of _cpuset, hence `set(cpuset)`. I will add a comment to clarify this